gpio-utils: Build failure

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

I am unable to build the `gpio-utils` package since commit f71277d66db862f2cef133e9d63cb858f38d0a0c

Entering the preserved build directory (`nix-build -K`), and running `make` with the above build flags reproduces the same error. Removing the `--eval=undefine` flag however seems to fix the issue. As such, this is likely related to #423933 - the build  works prior to this PR being merged. Any commits after this, including prior to and after the `gpio-utils` package being migrated to by-name in #435514 - produce an error related to empty variables from `make`.

### Steps to reproduce

1. Checkout revision `f71277d66db862f2cef133e9d63cb858f38d0a0c`.
2. Build the `pkgs/os-specific/linux/kernel/gpio-utils.nix` package (has since been moved to by-name in #435514, and can also be reproduced on the by-name version). I used `nix-build -E "let nixpkgs = import ../../../.. {}; in nixpkgs.callPackage ./gpio-utils.nix {}"` from the kernel directory.
3. Observe the build error
4. `get checkout HEAD^1` to go back one commit in history
5. Rerun the build
6. Observe build success

This can also be observed currently (at least for me) by trying to install a NixOS system from current nixpkgs master with a system configuration depending on `gpio-utils`.

### Expected behaviour

`gpio-utils` should build without errors

### Screenshots

N/A

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/p7bjcwb39ixlzdcz8vq33ng11irix2il-linux-6.12.42.tar.xz
source root is linux-6.12.42
separate-debug-info: recording checksum of source files for debug support...
setting SOURCE_DATE_EPOCH to timestamp 1755252854 of file "linux-6.12.42/virt/lib/irqbypass.c"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash O=\$\(buildRoot\) --eval=undefine modules ARCH=x86_64 CROSS_COMPILE=
make: *** empty variable name.  Stop.
```

### Additional context

_No response_

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.41, NixOS, 25.05 (Warbler), 25.05.20250813.3385ca0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/cfzjjbkm6p6whq10ha83dqv6j0ipp8fn-source`
```

### Notify maintainers


@kwohlfahrt 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

The first error (`make: *** empty variable name.  Stop.`) can be fixed with this patch:

```patch
--- a/pkgs/os-specific/linux/kernel/gpio-utils.nix
+++ b/pkgs/os-specific/linux/kernel/gpio-utils.nix
@@ -8,7 +8,8 @@ stdenv.mkDerivation {
   pname = "gpio-utils";
   version = linux.version;
 
-  inherit (linux) src makeFlags;
+  inherit (linux) src;
+  makeFlags = linux.commonMakeFlags;
 
   preConfigure = ''
     cd tools/gpio
```

That fixes the build until e9b2edb4916328647571aee01b2a83c02e585df5 (https://github.com/NixOS/nixpkgs/pull/402198) where it breaks again with this error:

```
/nix/store/dc9vaz50jg7mibk9xvqw5dqv89cxzla3-binutils-2.44/bin/ld: cannot find Scrt1.o: No such file or directory
/nix/store/dc9vaz50jg7mibk9xvqw5dqv89cxzla3-binutils-2.44/bin/ld: cannot find crti.o: No such file or directory
/nix/store/dc9vaz50jg7mibk9xvqw5dqv89cxzla3-binutils-2.44/bin/ld: cannot find -lgcc_s: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [Makefile:49: lsgpio] Error 1
```

--- a/pkgs/by-name/gp/gpio-utils/package.nix
+++ b/pkgs/by-name/gp/gpio-utils/package.nix
@@ -8,7 +8,10 @@ stdenv.mkDerivation {
   pname = "gpio-utils";
   version = linux.version;
 
-  inherit (linux) src makeFlags;
+  inherit (linux) src;
+  makeFlags = linux.commonMakeFlags ++ [
+    "CC=${stdenv.cc.targetPrefix}cc"
+  ];
 
   preConfigure = ''
     cd tools/gpio

